File: src/main/kotlin/models/JPA/Role.java
package models.JPA;


import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import org.springframework.validation.annotation.Validated;

import javax.persistence.*;

@Data
@Entity
@Validated
@Table(name="role")
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor
public class Role{

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "role_id")
    private Integer roleId;

    @Column(name = "role_name")
    private String role;

    public Role(String role){
        this.role = role;
    }
}

File: src/main/kotlin/models/jpa/Role.kt
package models.jpa

import javax.persistence.*

@Entity
@Table(name = "role")
data class Role(
    @Column(name = "role_name") val role: String,
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "role_id") val roleId: Int
)